packages/server/modules/activitystream/services/userActivity.ts

```diff
@@ -22,3 +40,38 @@ export const addUserUpdatedActivityFactory =
       message: 'User updated'
     })
   }
+
+const addUserDeletedActivityFactory =
+  (deps: { saveActivity: SaveActivity }) =>
+  async (params: { targetUserId: string; invokerUserId: string }) => {
+    const { targetUserId, invokerUserId } = params
+
+    await deps.saveActivity({
+      streamId: null,
+      resourceType: 'user',
+      resourceId: targetUserId,
+      actionType: ActionTypes.User.Delete,
+      userId: invokerUserId,
+      info: {},
+      message: 'User deleted'
+    })
+  }
+
+export const reportUserActivityFactory =
```

Comment:
This func is a valid listener initalization func, but i think it is not a service. This should be part of the application itself, next to the module init file somewhere.

I have not yet landed on a good pattern for this, but probably something like `eventListeners.ts` should be a good start

Reply:
altho i've done it myself, i don't like putting files next to the module init, it feels like a shortcut to avoid thinking about a proper place to actually put them, and then you end up with a bunch of unrelated, disorganized modules in the same location

i also don't want this to be a single file, cause you could have a ton of listeners of different types/concerns, especially in activitystream

I ended up moving this to /events/userListeners.ts

Reply:
that makes sense, as we are moving into utilizing events more and more